File: soal-bebas/soal-bebas.cpp
/*
  Nama File : soal-bebas.cpp
  Nama      : Andaru Danurdara Wibisana, Fauzan Azmi Dwicahyo, Irfan Kamal, Muhammad Attila An Naufal
  NPM       : 140810200020, 140810200030, 140810200032, 140810200048
  Kelas     : B
  Tanggal   : 13 Oktober 2021
  Deskripsi : Implementasi kode Permutasi dan Kombinasi dari pertemuan 4 pada C++
*********************************************/
#include <iostream>

class PersegiPanjang{
    private :
    float panjang, lebar;
    
    public :
    PersegiPanjang(){
        this->panjang = 0;
        this->lebar = 0;
    }
    
    PersegiPanjang(float panjang, float lebar){
        this->panjang = panjang;
        this->lebar = lebar;
    }
    
    void setPanjang(){
        std::cout << "Masukkan panjang persegi panjang  : ";
        std::cin >> this->panjang;
    }
    
    float getPanjang(){
        return this->panjang;
    }
    
    void setLebar(){
        std::cout << "Masukkan lebar persegi panjang    : ";
        std::cin >> this->lebar;
    }
    
    float getLebar(){
        return this->lebar;
    }
    
    float getKeliling(){
        return this->panjang+this->lebar;
    }
    
    float getLuas(){
        return this->panjang*this->lebar;
    }
};


int main(){
    PersegiPanjang persegiPanjang1;
    PersegiPanjang persegiPanjang2(float(6.5), float(3.0));
    persegiPanjang1.setPanjang();
    persegiPanjang1.setLebar();
            
    std::cout << "Panjang Persegi Panjang  1  : " << persegiPanjang1.getPanjang() << "\n";
    std::cout << "Lebar Persegi Panjang 1     : " << persegiPanjang1.getLebar() << "\n";
    std::cout << "Keliling Persegi Panjang 1  : " << persegiPanjang1.getKeliling() << "\n";
    std::cout << "Luas Persegi Panjang 1      : " << persegiPanjang1.getLuas() << "\n\n";

    std::cout << "Panjang Persegi Panjang  2  : " << persegiPanjang2.getPanjang() << "\n";
    std::cout << "Lebar Persegi Panjang 2     : " << persegiPanjang2.getLebar() << "\n";
    std::cout << "Keliling Persegi Panjang 2  : " << persegiPanjang2.getKeliling() << "\n";
    std::cout << "Luas Persegi Panjang 2      : " << persegiPanjang2.getLuas() << "\n\n";

    return 0;
}

File: soal-bebas/soal-bebas.py
# Nama File : soal-bebas.py
# Nama      : Andaru Danurdara Wibisana, Fauzan Azmi Dwicahyo, Irfan Kamal, Muhammad Attila An Naufal
# NPM       : 140810200020, 140810200030, 140810200032, 140810200048
# Kelas     : B
# Tanggal   : 13 Oktober 2021
# Deskripsi : Implementasi kode Permutasi dan Kombinasi dari pertemuan 4 pada C++

import math

class PersegiPanjang:
    def __init__(self, *args):
        if len(args) == 2:
            self.__panjang = float(args[0])
            self.__lebar = float(args[1])
        elif len(args) == 0:
            self.__panjang = float(0)
            self.__lebar = float(0)
    
    def setPanjang(self, panjang):
        self.__panjang = float(panjang)
    
    def getPanjang(self):
        return self.__panjang
    
    def setLebar(self, lebar):
        self.__lebar = float(lebar)
    
    def getLebar(self):
        return self.__lebar
    
    def getKeliling(self):
        return 2 * (self.__panjang + self.__lebar)
    
    def getLuas(self):
        return self.__panjang * self.__lebar

persegiPanjang1 = PersegiPanjang()
persegiPanjang2 = PersegiPanjang(float(6.5), float(3.0))

print("Masukkan panjang persegi panjang  : ", end='')
persegiPanjang1.setPanjang(input())
print("Masukkan lebar persegi panjang    : ", end='')
persegiPanjang1.setLebar(input())
        
print("Panjang Persegi Panjang  1  : ", persegiPanjang1.getPanjang())
print("Lebar Persegi Panjang 1     : ", persegiPanjang1.getLebar())
print("Keliling Persegi Panjang 1  : ", persegiPanjang1.getKeliling())
print("Luas Persegi Panjang 1      : ", persegiPanjang1.getLuas(), "\n")

print("Panjang Persegi Panjang  2  : ", persegiPanjang2.getPanjang())
print("Lebar Persegi Panjang 2     : ", persegiPanjang2.getLebar())
print("Keliling Persegi Panjang 2  : ", persegiPanjang2.getKeliling())
print("Luas Persegi Panjang 2      : ", persegiPanjang2.getLuas())
